pkg/secrets/secrets.go

```diff
@@ -80,3 +85,69 @@ func GenerateUserData(publicKey ssh.PublicKey) (*core.Secret, error) {
 
 	return userDataSecret, nil
 }
+
+// Encrypt creates an ecnrypted block of text from a plaintext message using the given key
+func Encrypt(plainText string, key []byte) (string, error) {
+	if key == nil {
+		return "", errors.New("encryption passphrase cannot be nil")
+	}
+
+	// prepare PGP block with a wrapper tag around the encrypted data
+	tag := "ENCRYPTED DATA"
+	msgBuffer := bytes.NewBuffer(nil)
+	encoder, _ := armor.Encode(msgBuffer, tag, nil)
+
+	writer, _ := openpgp.SymmetricallyEncrypt(encoder, key, nil, nil)
+	_, err := writer.Write([]byte(plainText))
+	if err != nil {
+		return "", err
+	}
+	writer.Close()
+	encoder.Close()
+
+	cipherText := string(msgBuffer.Bytes())
+	return cipherText, nil
+}
+
+// Decrypt conerts encrypted contents to plaintext using the given key as a passphrase
+func Decrypt(cipherText string, key []byte) (string, error) {
+	if key == nil {
+		return "", errors.New("decryption passphrase cannot be empty")
+	}
+
+	// Unwrap encoded block holding the message content
+	msgBuffer := bytes.NewBuffer([]byte(cipherText))
+	armorBlock, err := armor.Decode(msgBuffer)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to process given cipher %s", cipherText)
+	}
+
+	msgBody, err := readMessage(armorBlock.Body, key)
+	if err != nil {
+		return "", err
```

Comment:
can add a wrapper here too.

Reply:
Only if you are going to add more context and information to the error.